### PR TITLE
Angry cows code

### PR DESCRIPTION
**Pull request by HarryWangATX**

### CODE REVIEW

File: solutions/usaco-592.mdx [hunk 1]
```diff
@@ -69,4 +82,66 @@ int main() {
...
+			prev = next;
+			radius++;
+		}
+
+		// [start, ...., prev]
+		// ..., prev
+		return Math.abs(prev - start);
+	}
+
+	//CodeSnip
```
Comment (maggie-j-liu) [inline solutions/usaco-592.mdx:142]:
```suggestion
	//CodeSnip{Kattio}
```

File: solutions/usaco-592.mdx [hunk 2]
```diff
@@ -69,4 +82,66 @@ int main() {
...
+				break;
+			}
+
+			//Update our previous explosion and increment radius
+			prev = next;
+			radius++;
+		}
+
+		// [start, ...., prev]
+		// ..., prev
```
Comment (maggie-j-liu) [inline solutions/usaco-592.mdx:138]:
```suggestion
		//[start, ...., prev]
		//...., prev
```
for consistency (or you could add spaces to the other comments)

File: solutions/usaco-592.mdx [hunk 3]
```diff
@@ -69,4 +82,66 @@ int main() {
...
+	}
+
+	public static int getNum(int start, int direction) {
+		int radius = 1;
+		int prev = start;
+
+		while(true) {
+			int next = prev;
+			//Getting the furthest explosion index without exceeding radius
+			while(next + direction >= 0 && next + direction < N && Math.abs(positions[next + direction] - positions[prev]) <= radius) {
```
Comment (maggie-j-liu) [inline solutions/usaco-592.mdx:123]: i think this line is too long

File: solutions/usaco-592.mdx [hunk 4]
```diff
@@ -22,25 +22,36 @@ using namespace std;
...
 	int radius = 1;
 	int prev = start;
 
-	while(prev >= 0 && prev < N) {
+	while(true) {
 		int next = prev;
-		while(next + direction >= 0 && next + direction < N 
-				&& abs(positions[next + direction] - positions[prev]) <= radius) {
+		//Getting the furthest explosion index without exceeding the current radius
+		while(next + direction >= 0 && next + direction < N && abs(positions[next + direction] - positions[prev]) <= radius) {
```
Comment (maggie-j-liu) [inline solutions/usaco-592.mdx:37]: same, this also looks too long

File: solutions/usaco-592.mdx [hunk 5]
```diff
@@ -22,25 +22,36 @@ using namespace std;
 int N, A = 0;
 vector<int> positions;
 
+void setPrec() { cout << fixed << setprecision(15); }
+void unsyncIO() { cin.tie(0)->sync_with_stdio(0); }
+void setIn(string s) { freopen(s.c_str(),"r",stdin); }
+void setOut(string s) { freopen(s.c_str(),"w",stdout); }
```
Comment (maggie-j-liu) [inline solutions/usaco-592.mdx:28]: why add these if you're not going to use them?

**Review by maggie-j-liu — CHANGES_REQUESTED** [review]

File: solutions/usaco-592.mdx (hunk 5 above)
Comment (HarryWangATX) [reply solutions/usaco-592.mdx:28]: good question!

File: solutions/usaco-592.mdx [hunk 6]
```diff
@@ -26,17 +26,20 @@ int getNum(int start, int direction) {
 	int radius = 1;
 	int prev = start;
 
-	while(prev >= 0 && prev < N) {
+	while(true) {
 		int next = prev;
+		//Getting the furthest explosion index without exceeding the current radius
```
Comment (SansPapyrus683) [inline solutions/usaco-592.mdx:31]: `// Put a space after the // for comments`

File: solutions/usaco-592.mdx [hunk 7]
```diff
@@ -60,7 +63,9 @@ int main() {
 
 	sort(positions.begin(), positions.end());
 
-	for(int i = 0; i < N; i++) {
+
```
Comment (SansPapyrus683) [inline solutions/usaco-592.mdx:66]: two newlines? is this intended?
(i'd prefer 0 tbh)

File: solutions/usaco-592.mdx [hunk 8]
```diff
@@ -69,4 +74,65 @@ int main() {
...
 
 </CPPSection>
+
+<JavaSection>
+
+```java
+import java.util.*;
+import java.io.*;
+
+public class angrycowsB {
```
Comment (SansPapyrus683) [inline solutions/usaco-592.mdx:84]:
```suggestoin
public class Angry
```

File: solutions/usaco-592.mdx [hunk 9]
```diff
@@ -69,4 +74,65 @@ int main() {
...
+
+		io.println(A);
+		io.close();
+	}
+
+	public static int getNum(int start, int direction) {
+		int radius = 1;
+		int prev = start;
+
+		while(true) {
```
Comment (SansPapyrus683) [inline solutions/usaco-592.mdx:112]:
```suggestion
		while (true) {
```

File: solutions/usaco-592.mdx [hunk 10]
```diff
@@ -69,4 +74,65 @@ int main() {
...
+			int next = prev;
+			//Getting the furthest explosion index without exceeding radius
+			while(next + direction >= 0 && next + direction < N
```
Comment (SansPapyrus683) [inline solutions/usaco-592.mdx:115]: wait why put spaces after control flow statements up there but not down here
image

File: solutions/usaco-592.mdx [hunk 11]
```diff
@@ -69,4 +74,65 @@ int main() {
...
+			//We didn't find a new haybale, so the chain explosion is over
+			if(next == prev) {
+				break;
+			}
+
+			//Update our previous explosion and increment radius
+			prev = next;
+			radius++;
+		}
+
```
Comment (SansPapyrus683) [inline solutions/usaco-592.mdx:129]: 92, 96, 103, 124, and 129 are unnecessary

File: solutions/usaco-592.mdx (hunk 6 above)
Comment (SansPapyrus683) [reply solutions/usaco-592.mdx:31]: also space after the control statement

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
nitpicking time

File: solutions/usaco-592.mdx (hunk 11 above)
Comment (SansPapyrus683) [reply solutions/usaco-592.mdx:128]: just this and i think i can approve

File: solutions/usaco-592.mdx [hunk 12]
```diff
@@ -69,4 +72,62 @@ int main() {
...
+			}
+			// We didn't find a new haybale, so the chain explosion is over
+			if (next == prev) {
+				break;
+			}
+			// Update our previous explosion and increment radius
+			prev = next;
+			radius++;
+		}
+
```
Comment (SansPapyrus683) [inline solutions/usaco-592.mdx:124]: ok went over it again there's still some bad newlines
specifically 46, 56, 58, 62, 69 (also why is there no endl or \n there), 89, 93, 100, 108, and 124

File: solutions/usaco-592.mdx [hunk 13]
```diff
@@ -7,13 +7,12 @@ author: Qi Wang
 
 [Official Analysis](http://www.usaco.org/current/data/sol_angry_bronze_jan16.html)
```
Comment (ryanchou-dev) [inline solutions/usaco-592.mdx:8]:
```suggestion
[Official Analysis (Java)](http://www.usaco.org/current/data/sol_angry_bronze_jan16.html)
```

File: solutions/usaco-592.mdx [hunk 14]
```diff
@@ -26,47 +25,103 @@ int getNum(int start, int direction) {
...
 		prev = next;
 		radius++;
 	}
-
 	return abs(prev - start);
 }
 
-
 int main() {
 	cin.tie(0)->sync_with_stdio(0);
```
Comment (ryanchou-dev) [inline solutions/usaco-592.mdx:49]: not too good with c++, but would this be better?
```suggestion
	cin.tie(0) -> sync_with_stdio(0);
```

**Review by ryanchou-dev — COMMENTED** [review]
:)) nice code! just this minor fixes

File: solutions/usaco-592.mdx (hunk 14 above)
Comment (SansPapyrus683) [reply solutions/usaco-592.mdx:49]: no think the convention is no spaces for those arrows

**Review by SansPapyrus683 — APPROVED** [review]

**Review by ryanchou-dev — APPROVED** [review]
🥇

**Review by maggie-j-liu — APPROVED** [review]

**Review by brad182 — APPROVED** [review]

**Pull request merged**